scripts/filter_multimappers.py

```diff
@@ -0,0 +1,177 @@
+
+    parser.add_argument(
+        '-i', '--infile',
+        help="Path to the SAM input file, sorted by query name.",
+        type=Path
+        )
+
+    return parser
+
+
+def count_indels(alignment: pysam.AlignedSegment) -> int:
+    """Count the number of indels in an alignment based on its CIGAR string.
+
+    This function counts the number of indels in the alignment based on the
+    alignment CIGAR string returned by .cigartuples. Insertions are encoded as
+    1s whereas the deletions are encoded as 2.
+    Refer to the .cigartuples documentation for more information at:
+    https://pysam.readthedocs.io/en/v0.15.0/api.html#pysam.AlignedSegment.cigartuples
+
+    Args:
+        alignments(pysam.AlignedSegment): The read to count insertions for.
+
+
+    Returns:
+        int: The sum of insertions and deletions in the alignment.
+    """
+    alignment_indels: list[int] = [op[1] for op in alignment.cigartuples if op[0] == 1 or op[0] == 2]
+    indels: int = sum(alignment_indels)
+
+    return indels
+
+
+def find_best_alignments(alignments: List[pysam.AlignedSegment]) -> List[pysam.AlignedSegment]:
+    """Find alignments with less indels.
+
+    This function creates a list of tuples with the alignment object and its
+    number of indels. Then, computes the minimum number of indels and returns
+    a list with the alignments that have no more than that minimum number of
+    indels. In addition, it updates the tag 'NH' and 'XI' to match the final
+    number of alignments kept and its identifier respectively.
+
+    Args:
+        alignments (list[pysam.AlignedSegment]): alignments with the same query name
+
+    Retrns:
+        best_alignments (list[pysam.AlignedSegment]): alignments with the less indels
+    """
+    alignment_indels = [(aln, count_indels(alignment=aln)) for aln in alignments]
+    min_indels = min(alignment_indels, key=lambda x: x[1])[1]
+    best_alignments = [indelCount[0] for indelCount in alignment_indels if indelCount[1] == min_indels]
+    
+    for i in range(len(best_alignments)):
+        best_alignments[i].set_tag('NH', len(best_alignments))
+        best_alignments[i].set_tag('XI', i)
+
+    return best_alignments
+
+
+def write_output(alignments: List[pysam.AlignedSegment]) -> None:
+    """Write the output to the standard output (stdout).
+
+    Args:
+        alignments (list[pysam.AlignedSegment]): alignments with the same query name
+    """
+    if len(alignments) == 1:
+        sys.stdout.write(alignments[0].to_string() + '\n')
+    else:
+        for alignment in alignments:
+            sys.stdout.write(alignment.to_string() + '\n')
+
+
+def main(sam_file: Path) -> None:
+    """Filter multimappers by indels count.
+
+    Args:
+        sam_file (str): Path to the input SAM file.
+    """
+    with pysam.AlignmentFile(sam_file, "r") as samfile:
+
+        # Extract header from input SAM file and write it to stdout
+        header = samfile.header
+        sys.stdout.write(str(header))
```

Comment:
This is not a hard to understand code area, so no comment required (see my comment above). In fact, the code is just as easy to read as your comment. It would still be as easy to read (or even easier) and actually _shorter_ than your prose comment, if you put it all in one line:

```python
sys.stdout.write(str(samfile.header))
```